src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -218,6 +219,60 @@ public void resetMaxHealth_MaxHealthChanged_ResetsBackTo20()
 		assertEquals(20.0, player.getHealth(), 0);
 	}
 
+	@Test
+	public void kill_PLayerDeathEventDispatched()
+	{
+		TestPlugin plugin = MockBukkit.load(TestPlugin.class);
+		AtomicBoolean eventDispatched = new AtomicBoolean();
+		server.getPluginManager().registerEvents(new Listener() {
+			@EventHandler
+			public void onDeath(PlayerDeathEvent event)
+			{
+				eventDispatched.set(true);
+			}
+		}, plugin);
+
+		player.kill();
+		assertTrue(eventDispatched.get());
+	}
+
+	@Test
+	public void kill_NoKeepInventory_PlayerReset()
+	{
+		player.getWorld().setGameRule(GameRule.KEEP_INVENTORY, false);
+		player.kill();
+
+		assertEquals(0, player.getLevel());
+		assertEquals(0, player.getExp(), 0);
+		assertEquals(0, player.getFoodLevel());
+		assertTrue(player.getInventory().isEmpty());
+		assertTrue(player.getOpenInventory() instanceof SimpleInventoryViewMock);
+	}
+
+	@Test
+	public void kill_KeepInventory_inventoryKept()
+	{
+		player.getWorld().setGameRule(GameRule.KEEP_INVENTORY, true);
+		player.getInventory().addItem(new ItemStack(Material.DIRT));
+		player.kill();
+
+		assertFalse(player.getInventory().isEmpty());
+	}
```

Comment:
Not sure if this belongs here.
We can unit-test gamerules elsewhere in my opinion.
I'd be fine removing this single test for now and worry about it later, since I was gonna rewrite the whole gamerule mocking anyway.

Reply:
This case tests a specific behavior of the tested method, I don't think it is a good idea to remove it. The other way around would be to listen to the event and disable `keep inventory` from there